src/main/java/mekanism/common/tile/TileEntityTeleporter.java

```diff
@@ -200,14 +232,24 @@ private void teleport() {
         TileEntityTeleporter teleporter = MekanismUtils.getTileEntity(TileEntityTeleporter.class, teleWorld, closestPos);
         if (teleporter != null) {
             for (Entity entity : getToTeleport()) {
-                teleporter.didTeleport.add(entity.getUniqueID());
+                entity.getSelfAndPassengers().forEach(e -> teleporter.didTeleport.add(e.getUniqueID()));
                 teleporter.teleDelay = 5;
                 teleportEntityTo(entity, closestCoords, teleporter);
                 if (entity instanceof ServerPlayerEntity) {
                     alignPlayer((ServerPlayerEntity) entity, closestPos);
                 }
                 for (Coord4D coords : getFrequency(FrequencyType.TELEPORTER).getActiveCoords()) {
-                    BlockPos coordsPos = coords.getPos();
+                    TileEntity tileEntity = world.getTileEntity(coords.getPos());
```

Comment:
Don't use `world#getTileEntity`, use `MekanismUtils#getTileEntity` it has extra checks to prevent ghost loading of chunks and also can validate tile types if needed.

Reply:
ok